src/main/java/com/timgroup/statsd/AlphaNumericMessage.java

```diff
@@ -0,0 +1,66 @@
+package com.timgroup.statsd;
+
+import java.util.Objects;
+
+
+public abstract class AlphaNumericMessage extends Message {
+
+    protected String value;
+
+    protected AlphaNumericMessage(Message.Type type) {
+        super(type);
+    }
+
+    protected AlphaNumericMessage(String aspect, Message.Type type, String value, String[] tags) {
+        super(aspect, type, tags);
+        this.value = value;
+    }
+
+    /**
+     * Aggregate message.
+     *
+     * @param message
+     *     Message to aggregate.
+     */
+    @Override
+    public void aggregate(Message message) { }
+
+    /**
+     * Get underlying message value.
+     *
+     * @return returns the value for the Message
+     */
+    public String getValue() {
+        return this.value;
+    }
+
+    @Override
+    public int hashCode() {
+
+        // cache it
+        if (this.hash == null) {
+            super.hashCode();  // will instantiate hash if null
+            if (this.value != null) {
+                this.hash += Objects.hash(this.value);
```

Comment:
I did, but I did not want to introduce a new library like apache commons or guava. Besides, the `Objects.hash()` implementation is decent. That said, you did make me realize there were a few things I had to a better job of:
- Not all attributes we were hashing were final. As you can see from the implementation this is critical, because we cache the hash code, so we need to guarantee that hash will not mutate (ie. the attributes used to calculate shouldn't either).
- The way we're combining the hashes here isn't great, it was just a dumb sum. We need to shift, then sum, so I've added that logic which was borrowed from the official `Arrays.hashCode` implementation.

Things should be better now. Thank you for showing your concern for this, we're in a better place now. 🙇